### PR TITLE
[IMP] mail: bounce email going directly to catchall when they are no …

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
Task: https://www.odoo.com/web#id=37971&view_type=form&model=project.task&action=333&active_id=965&menu_id=4720

### CODE REVIEW

File: addons/mail/models/mail_thread.py [hunk 1]
```diff
@@ -1134,6 +1135,18 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
...
             # no route found for a matching reference (or reply), so parent is invalid
             message_dict.pop('parent_id', None)
+
+            # check it does not directly contact catchall
+            if catchall_alias and catchall_alias in email_to_localpart:
+                _logger.info('Routing mail from %s to %s with Message-Id %s: direct write to catchall, bounce', email_from, email_to, message_id)
+                body = """<div>
+<p>Hello,</p>
+<p>The following email sent to %s cannot be accepted because this is a catchall email
+   used to collect replies. BLAH BLAH BLAH</p>
```
Comment (bst-odoo) [inline addons/mail/models/mail_thread.py]: blah blah blah ?

Comment (tde-banana-odoo) [reply addons/mail/models/mail_thread.py]: Yeah. I wonder if boom boom boom boom boom if not better.

**Comment by wtaferner** [comment]
@tde-banana-odoo 
It is safe to backport this?

**Comment by tde-banana-odoo** [comment]
Hello,

Well at first sight I would say you have to protect yourself against the new template not being present in the database (mail.mail_bounce_catchall) unless the mail module is updated.

A method has its API that changes (_routing_create_bounce_email) but as it is an internal method it should not be inherited. In official addons there is no override of this method.

Remember I am a dev for master, not someone working on stable or even production instances, so I will let you ask it to smart people :) .

**Comment by wtaferner** [comment]
Haha, I think I have already the most qualified answer to my question. 

I summarize as following: 
If you do it right for your instances it is safe if you have no clue what you are doing please do not use it in stable ;-)

Do you agree with my translation? ;-)

**Comment by tde-banana-odoo** [comment]
> Hello @tde-banana-odoo , this PR is so old, but may I ask why we need to bounce emails that send directly to catchall?

Because it is used to collect replies to existing threads only. New emails should always target aliases that define the action to do with it. Trying to do everything on a single alias is error prone, better separate things. It acts a bit like a "no reply" actually.

**Pull request merged**